src/interactive/azext_interactive/azclishell/chatgpt_generate.py

```diff
@@ -0,0 +1,73 @@
+from enum import Enum
+import threading
+import json
+from ._azclierror import ChatgptGenrateError
+from .recommendation import RecommendType
+
+
+class ChatgptThread(threading.Thread):
+    def __init__(self, cli_ctx, user_msg, search_path, executing_command, on_prepared_callback, history_msg=None):
+        super().__init__()
+        self.result = None
+        self.cli_ctx = cli_ctx
+        self.user_msg = user_msg
+        self.history_msg = history_msg
+        self.on_prepared_callback = on_prepared_callback
+        # maintain a copy of the command history to insert commands executed in the searched scenario
+        self.command_history = search_path.get_cmd_history(25)
+        if executing_command:
+            self.command_history.append(executing_command)
+        self.processed_exception = search_path.get_result_summary() if not executing_command else None
+        self.api_version = None
+
+    def run(self) -> None:
+        try:
+            result, self.api_version = generate_script(user_msg=self.user_msg, history_msg=self.history_msg)
+            self.result = {"content": transform_script_to_scenario(result["content"]),
+                           "history_msg": result["history_msg"], "api_version": self.api_version,
+                           "type": RecommendType.Chatgpt}
+        except ChatgptGenrateError:
+            self.result = {"content": "Connection Error. Please check your network connection.",
+                           "api_version": self.api_version, "type": RecommendType.Error}
+
+
+def generate_script(user_msg: str, history_msg: list):
+    """Generate CLI Scripts with ChatGPT model"""
+    import requests
+    url = "https://cli-recommendation.azurewebsites.net/api/chatgptservice"
+    payload = {
+        "user_msg": user_msg,
+        "history_msg": history_msg
+    }
+    try:
+        response = requests.post(url, json.dumps(payload))
+        response.raise_for_status()
+    except requests.ConnectionError as e:
+        raise ChatgptGenrateError(f'Network Error: {e}') from e
+    except requests.exceptions.HTTPError as e:
+        raise ChatgptGenrateError(f'{e}') from e
+    except requests.RequestException as e:
+        raise ChatgptGenrateError(f'Request Error: {e}') from e
+
+    result = None
+    if 'data' in response.json():
+        # result.content: the generated script
+        # result.history_msg: the history of the conversations
+        result = response.json()['data']
+    api_version = None
+    if 'api_version' in response.json():
+        api_version = response.json()['api_version']
+    return result, api_version
+
+
+def transform_script_to_scenario(script):
+    """Transform the generated script to a scenario"""
+    # source 5 means it is from chatgpt service
+    scenario = {'scenario': script['Description'], 'nextCommandSet': script['CommandSet'], 'source': 5, 'type': RecommendType.Chatgpt,
```

Comment:
> 'source': 5

Could we use enumeration values instead of `5`?

Reply:
There are no enumerate sets in the az interactive extension to show the type source, it is a property given from the service, maybe we can modify the response data to change it.